alcs-api/src/application/application-code/application-code.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { ApplicationDecisionMaker } from './application-decision-maker/application-decision-maker.entity';
+import { ApplicationStatus } from '../application-status/application-status.entity';
+import { ApplicationType } from './application-type/application-type.entity';
+
+@Injectable()
+export class ApplicationCodeService {
+  constructor(
+    @InjectRepository(ApplicationType)
+    private typeRepository: Repository<ApplicationType>,
+    @InjectRepository(ApplicationStatus)
+    private statusRepository: Repository<ApplicationStatus>,
+    @InjectRepository(ApplicationDecisionMaker)
+    private decisionMakerRepository: Repository<ApplicationDecisionMaker>,
+  ) {}
+
+  async getAllCodes() {
+    const values = await Promise.all([
+      this.typeRepository.find(),
+      this.statusRepository.find(),
+      this.decisionMakerRepository.find(),
+    ]);
```

Comment:
So two part mapping to get codes? One to stick it in the service and one to explicitly expose it on the controller?

Reply:
Yes. Services should be the ones interacting with the DB and controllers handling mapping to the front end.

Reply:
I could probably cache the ones that are static, e.g. everything but status, and load them once when the service starts.